### PR TITLE
[LibGit2_jll] Update to v1.6.1

### PR DESCRIPTION
**Pull request by fxcoudert**
Depends on https://github.com/JuliaPackaging/Yggdrasil/pull/6297

~Could someone remind me of the magic command to regenerate `deps/checksums/libgit2`?~
got it: `make -f contrib/refresh_checksums.mk libgit2`

### CODE REVIEW

File: deps/libgit2.version [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
-LIBGIT2_SHA1=fbea439d4b6fc91c6b619d01b85ab3b7746e4c19
+LIBGIT2_BRANCH=v1.6.0
+LIBGIT2_SHA1=f7325c448053bfcd1a32ec319ed392f525480634
 
 ## Other deps
 # Specify the version of the Mozilla CA Certificate Store to obtain.
 # The versions of cacert.pem are identified by the date (YYYY-MM-DD) of their changes.
 # See https://curl.haxx.se/docs/caextract.html for more details.
-MOZILLA_CACERT_VERSION := 2022-10-11
+MOZILLA_CACERT_VERSION := 2023-01-10
```
Comment (giordano) [inline deps/libgit2.version:13]: We probably want to update it on Yggdrasil (https://github.com/JuliaPackaging/Yggdrasil/blob/852acea4fd0bdc48ed72601e73ff4e087d436b6c/M/MozillaCACerts/build_tarballs.jl) and then here https://github.com/JuliaLang/julia/blob/f0eadd076f2728797cf5be88b61cda1d9f1dd515/stdlib/MozillaCACerts_jll/Project.toml#L3

Comment (giordano) [reply deps/libgit2.version:13]: Ok, I updated the certs in Yggdrasil (https://github.com/JuliaPackaging/Yggdrasil/pull/6310), so now you can update the version number in `Project.toml` and then referesh the checksums with `make -f contrib/refresh_checksums.mk mozillacert` :slightly_smiling_face:

**Comment by fxcoudert** [comment]
Updated and pushed

File: stdlib/LibGit2_jll/src/LibGit2_jll.jl [hunk 2]
```diff
@@ -21,9 +21,9 @@ libgit2_path = ""
 if Sys.iswindows()
     const libgit2 = "libgit2.dll"
 elseif Sys.isapple()
-    const libgit2 = "@rpath/libgit2.1.5.dylib"
+    const libgit2 = "@rpath/libgit2.1.6.dylib"
 else
-    const libgit2 = "libgit2.so.1.5"
+    const libgit2 = "libgit2.so.1.6"
```
Comment (giordano) [inline stdlib/LibGit2_jll/src/LibGit2_jll.jl:26]: Sounds like libgit2 v1.6 has still soversion 1.5.  Not sure that was done on purpose though, although it'd be great if they didn't break the  ABI again.

Comment (giordano) [reply stdlib/LibGit2_jll/src/LibGit2_jll.jl:26]: Might have not been done on purpose, given the fact now there's v1.6.1 which includes this: https://github.com/libgit2/libgit2/commit/8a871d13b7f4e186b8ad943ae5a7fcf30be52e67

Comment (giordano) [reply stdlib/LibGit2_jll/src/LibGit2_jll.jl:26]: And [v1.6.0](https://github.com/libgit2/libgit2/releases/tag/v1.6.0) was retracted

Comment (inkydragon) [reply stdlib/LibGit2_jll/src/LibGit2_jll.jl:26]: It looks like the build product is still called `libgit2.so.1.5`
https://github.com/JuliaBinaryWrappers/LibGit2_jll.jl/releases/tag/LibGit2-v1.6.0%2B0

Comment (giordano) [reply stdlib/LibGit2_jll/src/LibGit2_jll.jl:26]: Yeah, they tagged 1.6.0 ***before*** merging https://github.com/libgit2/libgit2/pull/6501, the soversion was not bumped in v1.6.0.  This has been rectified in v1.6.1.

**Comment by fxcoudert** [comment]
New Yggdrasil PR for 1.6.1: https://github.com/JuliaPackaging/Yggdrasil/pull/6313

**Comment by giordano** [comment]
The rebase didn't go well, this PR now includes unrelated commits

**Comment by fxcoudert** [comment]
Should be fixed now

**Comment by fxcoudert** [comment]
Three failures:

Capture d’écran 2023-02-26 à 17 17 31

I do not think they are related.

**Comment by giordano** [comment]
I restarted the macOS job, second time worked like a charm. FreeBSD and PowerPC are known to have problems (and for this reason they are [tier 2 or 3](https://julialang.org/downloads/#supported_platforms)), and they are allowed to fail tests in CI, in fact they always fail. Thanks a lot for your work!

**Pull request merged**